#MA705 Yelp Report - Related Variables of Interest  (spring 2019, codingwatchcity)

- CodingWatchCity
- Spring 2019

##Team Members:
Yanru Yang: yang_yanr@bentley.edu    
Xinyu Ma: ma_xiny@bentley.edu    
Shuangyu Jia: jia_shua@bentley.edu   
Weiran Lyu: lyu_weir@bentley.edu   
Naili Zhu: zhu_nail@bentley.edu

## Introduction   
In notebook **0. Contents**, we specified our dataset as `yelp` and selected tables `business` and `review`. In this notebook, the main objectives are to explore and clean our dateset, and to transform variables to tables, views and dataframes, which can be used direclty.

## Contents 
1. Objects Selection
2. Join Table
3. Final Dataset
4. Common views

## 1. Objects Selection

In [6]:
%sql
select name, count(business_id) from yelp.business
group by name
order by count(business_id) desc

name count(business_id) Starbucks 1026 McDonald's 782 Subway 743 Pizza Hut 318 Tim Hortons 315 Taco Bell 305 Walgreens 303 Burger King 291 Wendy's 281 The UPS Store 273 Domino's Pizza 233 Great Clips 231 Circle K 223 Enterprise Rent-A-Car 216 Dunkin' Donuts 188 KFC 185 Chase Bank 181 Chipotle Mexican Grill 180 Wells Fargo Bank 171 Dairy Queen 165 Bank of America 165 7-Eleven 159 Panera Bread 159 Jimmy John's 153 Supercuts 151 Papa John's Pizza 151 Panda Express 149 T-Mobile 143 Popeyes Louisiana Kitchen 139 Arby's 135 US Post Office 133 Shoppers Drug Mart 122 The Home Depot 119 Denny's 117 CVS Pharmacy 115 Target 115 QuikTrip 110 Walmart Supercenter 109 Firestone Complete Auto Care 108 Little Caesars Pizza 106 PetSmart 105 Jiffy Lube 104 Anytime Fitness 104 Cvs Pharmacy 102 Five Guys 100 Dollar Tree 98 Pizza Pizza 98 Second Cup 96 Public Storage 96 Sonic Drive-In 95 Jersey Mike's Subs 93 Giant Eagle 91 Swiss Chalet Rotisserie & Grill 89 Chick-fil-A 89 Discount Tire 87 Best Buy 87 FedEx Office Print & Ship Center 87 Walmart 86 Firehouse Subs 86 Jack in the Box 84 GoodLife Fitness 81 Verizon 81 Sally Beauty Supply 80 Hertz Rent A Car 79 AT&T 79 Midas 79 Leslie's Pool Supplies, Service & Repair 78 Ross Dress for Less 74 IHOP 73 Staples 73 Applebee's Grill + Bar 73 Sprint Store 71 Boston Pizza 71 AutoZone 71 Michaels 70 LA Fitness 69 U.S. Bank 68 Filiberto's Mexican Food 67 Papa Murphy's 67 Banfield Pet Hospital 66 Goodwill 66 Cold Stone Creamery 66 Buffalo Wild Wings 66 Del Taco 64 Autozone 63 Meineke Car Care Center 63 Canadian Tire 61 Thai Express 61 Pep Boys 60 Albertsons 60 Tim Horton's 60 Jack In the Box 59 LensCrafters 59 Gamestop 59 Einstein Bros Bagels 59 A&W 58 Harris Teeter 58 Petco 58 Terrible Herbst 57 Jamba Juice 57 LCBO 55 Extra Space Storage 55 Port of Subs 55 Kohl's 54 Safeway 54 Bath & Body Works 53 Shell 53 Chevron 53 Rite Aid 53 O'Reilly Auto Parts 53 Sephora 52 Roberto's Taco Shop 52 Marco's Pizza 51 Edible Arrangements 51 El Pollo Loco 50 Chili's 50 Harvey's 49 Life Storage 48 Freshii 48 Ulta Beauty 48 Tropical Smoothie Cafe 48 H&R Block 47 Smashburger 47 Pizza Nova 47 Sonora Quest Laboratories 46 Sunset Grill 46 Dollarama 45 Olive Garden Italian Restaurant 45 Wild Wing 45 European Wax Center 45 Wingstop 45 Party City 44 Bojangles' Famous Chicken 'n Biscuits 44 Valvoline Instant Oil Change 44 Marshalls 44 Quiznos 44 Capriotti's Sandwich Shop 43 Church's Chicken 43 Bruegger's Bagels 43 Outback Steakhouse 42 Carl's Jr 42 Metro 42 Bob Evans 42 Aldi 41 Hungry Howie's Pizza 41 Moe's Southwest Grill 40 Monro Muffler Brake & Service 40 Red Lobster 40 99 Cents Only Stores 39 Costco 39 Carl's Jr. 39 Bed Bath & Beyond 39 NTB - National Tire & Battery 39 Verizon Authorized Retailer - GoWireless 39 Booster Juice 39 Waffle House 38 Sears 38 Barro's Pizza 38 Lowe's Home Improvement 38 Fantastic Sams Hair Salons 38 The Joint Chiropractic 38 CubeSmart Self Storage 38 Mucho Burrito 38 Aroma Espresso Bar 38 Victoria's Secret 38 Long John Silver's 37 Nationwide Vision Centers 37 Winners 36 Eggsmart 36 Walmart Neighborhood Market 36 SmartStyle 36 GNC 36 Smoothie King 36 Sobeys 36 Banner Urgent Care 36 Avis Rent A Car 36 Men's Wearhouse 36 Visionworks 36 Old Navy 36 Fresh & Easy 35 QDOBA Mexican Eats 35 Sheetz 35 Motel 6 35 SuperPawn 35 Culver's 35 Red Robin Gourmet Burgers 35 PostNet 35 Baskin-Robbins 34 Quest Diagnostics 34 The Coffee Bean & Tea Leaf 34 Palm Beach Tan 34 Little Caesars 34 Big 5 Sporting Goods 34 Rosati's Pizza 34 Boston Market 34 Sherwin-Williams Paint Store 33 Sprouts Farmers Market 33 Pita Pit 33 In-N-Out Burger 33 First Watch 33 uBreakiFix 32 Toys R Us 32 Kmart 32 RadioShack 32 Peter Piper Pizza 32 Marc's Stores 32 Pier 1 Imports 32 FastMed Urgent Care 32 Tuesday Morning 32 Big O Tires 31 Food Lion 31 Sam's Club 31 Regal Nails 31 Lowe's 31 Mattress Firm Clearance 31 Banana Republic 31 Cost Cutters 31 Pizzaiolo 31 Menchie's Frozen Yogurt 31 St. Louis Bar & Grill 30 Batteries Plus Bulbs 30 Eat'n Park 30 Macy's

By using SQL `select` and `group by`, we check the number of stores of every brand. In addtion, we use `order by` and `desc` to arrange the results in a descending order.   
   
From the output above, we find that Starbucks has the largest records, so we decide to focus on data of Starbucks. Starbucks ranks as the top business with 1,026 stores in `yelp` dataset.

In [8]:
%sql
select name, count(business_id) from yelp.business
where name Like "%tarbuck%"
group by name
order by count(business_id) desc

name count(business_id) Starbucks 1026 Starbucks Coffee 8 starbucks 2 Williams & Starbuck 1 Starbucks Atrium 1 Starbucks - Albertsons 1 Starbuck 1 Starbucks Reserve 1 Starbucks at Mandarin 1

After grouping `business` by name, we realize that there are different spellings for the same business. For example, Starbucks can be written as "Starbucks", "starbucks", " Starkbucks Coffee" and so on. To keep data integrity, we decide to use the `like` operator to retrieve all the businesses which names are similar to Starbucks and keep all of them for further research. The coding in cmd 6 only gives the number of stores that use exactly "Starbucks" as name. After adding all stores with similar names, the number of observations becomes 1,042.

In [10]:
%sql
drop view if exists business_new;
create temporary view business_new
as select * from yelp.business
where name Like "%tarbuck%"

Since we only focus on Starbucks, we create a new temporary view for `business` table, which contains the only information for Starbucks. We name the new temporary view as `business_new`.

## 2. Join table   
We focus on `business` and `review` table. After doing a preliminary analysis on all the variables in these two tables, we decided to join these two tables, so that we can know more about the relation between a business and its review results. It will help us achieve our research goal.

**(1) Join Table**

In [14]:
%sql
/* Combine business and review*/
drop view if exists combine;
create temporary view combine
as select yelp.business.business_id, yelp.business.name, yelp.business.stars, yelp.business.is_open, yelp.business.city, yelp.business.hours,yelp.business.state, yelp.business.latitude, yelp.business.longitude, yelp.business.neighborhood, yelp.business.attributes, yelp.review.stars as review_stars, yelp.review.text, yelp.review.date, yelp.review.user_id, yelp.review.useful, yelp.review.review_id
from yelp.business
left join  yelp.review
on yelp.business.business_id == yelp.review.business_id

We use SQL code to create the temperary view named `combine` by joining the business and the review tables. The `business` table is left joined with the `review` table, setting `business_id` as the matching criteria. Outer join is not performed because our study focuses on Starbucks with applicable business information.

**(2) Filter Starbucks Records**

In [17]:
%sql
/* Create table for Starbucks */
drop table if exists starbucks;
create temporary view starbucks
as select *
from combine
where name Like "%tarbuck%"

Since we target only on Starbucks, we creat temporary view `starbarks` through using the SQL operator `like`.

In [19]:
%sql
select count(*) from starbucks

count(1) 22253

We use SQL `select` command to display all rows and columns. We also count the number of rows. The temporary `starbucks` view includes 17 columns and 22,253 rows.

## 3. Final Dataset   
Our final dataset includes a table `review` in yelp, and two temporary views `business_new` and `starbucks`.

##4.Common Views and Dataframes
The views below are created for the convenience. They have been created and interpreted in other notebooks.

**(1) Major view created from `business`**   
According to our data cleaning work, we create `business_new` view to store all the Starbucks information in original `business` table, which can be refered to cmd 10.

**(2) Create `busi_df` Python Pandas Dataframe**   
We transfer the SQL temporary view `business_new` into Python Pandas Dataframe, which can be used as a basic dataframe in Python.

In [25]:
%python
# business_df : pandas dataframe of business file
busi_df = spark.sql('select * from business_new').toPandas()

**(3) Major view for all combined information from `business` and `review` table**   
`starbucks` view has already created in cmd 17.

**(4) Create `starb_df` Python Pandas Dataframe**   
We transfer the SQL temporary view `starbucks` into the python pandas dataframe, which can be used as a basic dataframe in Python.

In [28]:
%python
starb_df = spark.sql('select * from starbucks').toPandas()

**(5) Create `busi_state_df` Pandas dataframe**   
We select state information from `business_new` view, then store it as a pandas dataframe.

In [30]:
%python
busi_state_df= spark.sql ('select state from business_new').toPandas()

**(6) Create `busi_revcount_df` Pandas dataframe**   
We select review_count information from `business_new` view, then store it as a pandas dataframe.

In [32]:
%python
busi_revcount_df= spark.sql ('select review_count from business_new').toPandas()

**(7) Create `busi_city_df` Pandas dataframe**   
We select state information from `business_new` view, then store it as a pandas dataframe.

In [34]:
%python
busi_city_df= spark.sql ('select city from business_new').toPandas()

**(8) Create `Hour` sql view and `Hour_df` Pandas dataframe.**

In [36]:
%sql
drop view if exists Hour;
create temporary view Hour
as select business_id,state, stars, split(hours.Saturday,"-")[0] as satopen, split(hours.Saturday,"-")[1] as satend,split(hours.Sunday,"-")[0] as sunopen,split(hours.Saturday,"-")[1] as sunend from business_new

In [37]:
%python
#assume none as close all day
Hour_df = spark.sql("select business_id,satopen,satend,sunopen,sunend from Hour").toPandas().fillna(value="0:0")


**(9) `Hour` data transformation.**

In [39]:
%python
# transfer string to datetime
import pandas as pd
  
Hour_df["satopen"]= pd.to_datetime(Hour_df['satopen'], format='%H:%M')
Hour_df["satend"]=pd.to_datetime(Hour_df['satend'], format='%H:%M')
Hour_df["sunopen"]=pd.to_datetime(Hour_df['sunopen'], format='%H:%M')
Hour_df["sunend"]=pd.to_datetime(Hour_df['sunend'], format='%H:%M')


#calculate time difference
import numpy as np
Hour_df['satdiff'] = Hour_df.satend-Hour_df.satopen
Hour_df['satdiff'] = Hour_df['satdiff']/np.timedelta64(1,'h')
Hour_df['sundiff'] = Hour_df.sunend-Hour_df.sunopen
Hour_df['sundiff'] = Hour_df['sundiff']/np.timedelta64(1,'h')
#calculate time difference over 24 hours
for i in range(len(Hour_df["satdiff"])):
  if Hour_df.iloc[i,5]<0:
     Hour_df.iloc[i,5]= Hour_df.iloc[i,5]+24

for i in range(len(Hour_df["sundiff"])):
  if Hour_df.iloc[i,6]<0:
     Hour_df.iloc[i,6]= Hour_df.iloc[i,6]+24

#Extract time from Datetime 
Hour_df['satopen'] =  pd.DatetimeIndex(Hour_df['satopen']).time
Hour_df['satend'] =  pd.DatetimeIndex(Hour_df['satend']).time
Hour_df['sunopen'] =  pd.DatetimeIndex(Hour_df['sunopen']).time
Hour_df['sunend'] =  pd.DatetimeIndex(Hour_df['sunend']).time

Hour_df

Out[ 7 ]: 
 business_id satopen satend ... sunend satdiff sundiff
0 OyJDaAAMr220qkZsovCARQ 04:30:00 20:00:00 ... 20:00:00 15.5 15.5
1 FGQZSl-P-SjMRmDyNNQUvw 00:00:00 00:00:00 ... 00:00:00 0.0 0.0
2 7DBu4i4sN9I8HWS3YaxArA 05:00:00 20:00:00 ... 20:00:00 15.0 15.0
3 ZQ-7uFQk21NHoOzJfhEjBw 04:30:00 21:00:00 ... 21:00:00 16.5 15.5
4 Mq6qiz-Bxgd8VE6tBwCuUg 07:00:00 23:00:00 ... 23:00:00 16.0 16.0
5 n947e5vGkuk7_qwzeNOdMg 06:00:00 23:00:00 ... 23:00:00 17.0 17.0
6 _LufK-TTQYtzmkLh41xvSA 07:00:00 23:00:00 ... 23:00:00 16.0 16.0
7 LtbuQJJn9pRhNf9BqGuNBQ 06:00:00 23:00:00 ... 23:00:00 17.0 17.0
8 Y1WPRHUR2goocvJcNsBlYA 06:30:00 22:00:00 ... 22:00:00 15.5 15.5
9 0rg8sJGPWebnxNFExpNdHA 07:00:00 17:00:00 ... 17:00:00 10.0 8.0
10 D2y7jMJiU5abpRicNpoCww 05:00:00 20:00:00 ... 20:00:00 15.0 15.0
11 7ITv2jv4rB-UuQwAo2ntjw 08:00:00 21:00:00 ... 21:00:00 13.0 13.0
12 xedxfb3rfEmXDE0BczcCBA 04:30:00 22:30:00 ... 22:30:00 18.0 18.0
13 vombhlVsSNWZEJ1xJ3jbbg 00:00:00 00:00:00 ... 00:00:00 0.0 0.0
14 WTv4tytSf7b_rdjyoFPDYw 07:00:00 14:30:00 ... 14:30:00 7.5 7.5
15 7sakqP2av046jujj-ERHhg 00:00:00 00:00:00 ... 00:00:00 0.0 0.0
16 s7Cq3DdSELAU_IFYXOnicg 08:00:00 21:00:00 ... 21:00:00 13.0 13.0
17 3BZIZF3O3QQmfjTYy5wDOg 05:00:00 20:00:00 ... 20:00:00 15.0 15.0
18 ywH06bIFXJ5q-H5BGHGh8A 00:00:00 00:00:00 ... 00:00:00 0.0 0.0
19 Ra7y4-gQvEhWfc-xrwi7Cw 07:00:00 20:30:00 ... 20:30:00 13.5 13.0
20 bD5sbDj_lMYnyUpbG3xm3Q 06:00:00 23:00:00 ... 23:00:00 17.0 17.0
21 ziTL7YYUZykEWzGEvO5Gsw 00:00:00 00:00:00 ... 00:00:00 0.0 0.0
22 zm-nB9xWL0RWZ-zoL7JNuQ 05:30:00 21:00:00 ... 21:00:00 15.5 15.0
23 QSfhaQD4sJ18oVwnwae4Kg 05:00:00 19:00:00 ... 19:00:00 14.0 10.5
24 H6W6detG0U5-QYGs5MItyw 04:30:00 22:00:00 ... 22:00:00 17.5 17.5
25 PxNlfFHLRVGKqpN7-vXBTQ 00:00:00 00:00:00 ... 00:00:00 0.0 0.0
26 SCb1mubW3RTIBfycjrql7A 00:00:00 00:00:00 ... 00:00:00 0.0 0.0
27 dl5Pz0gAP47SpCZ_eh_uQA 05:00:00 23:00:00 ... 23:00:00 18.0 18.0
28 NdXVQ6-36v68UnwtLPjx7Q 04:30:00 23:00:00 ... 23:00:00 18.5 18.0
29 q0q4Y3CswRvl0XflTQy9IQ 05:30:00 23:00:00 ... 23:00:00 17.5 16.5
... ... ... ... ... ... ... ...
1012 iyJCmE19VV6ttawCioIF0A 05:00:00 19:30:00 ... 19:30:00 14.5 14.5
1013 vYu1_HnKIcj6_rhw3p4qYw 07:00:00 21:00:00 ... 21:00:00 14.0 14.0
1014 oUEtUOz1ovJ9lqFoRA4aKw 06:00:00 22:00:00 ... 22:00:00 16.0 15.0
1015 1kUjJHO_Kwthd7jCDozuYg 05:00:00 20:00:00 ... 20:00:00 15.0 15.0
1016 WCl3rZyr2fmNX6KJLsTg4A 00:00:00 00:00:00 ... 00:00:00 0.0 0.0
1017 KgcKoYiZdwFJ158Gx3UJyQ 00:00:00 00:00:00 ... 00:00:00 0.0 0.0
1018 RmTjXHc2wF645_XlCo2ofw 05:00:00 19:30:00 ... 19:30:00 14.5 14.5
1019 bbQblTw9q18TYQHuIfxsng 00:00:00 00:00:00 ... 00:00:00 0.0 0.0
1020 XAgJSBOBGw4_qqRsO60CNQ 06:00:00 22:30:00 ... 22:30:00 16.5 16.5
1021 xXMEKQZS3HU-TnHvC51aOw 07:00:00 21:00:00 ... 21:00:00 14.0 14.0
1022 wuCyYFZKRIj7Q0m44_o_qQ 06:00:00 20:00:00 ... 20:00:00 14.0 14.0
1023 9p2phMmjGq2QJ6PNO9LZxA 07:00:00 23:00:00 ... 23:00:00 16.0 16.0
1024 CtI7MMRF-arzGJxA1nJccQ 05:00:00 21:00:00 ... 21:00:00 16.0 16.0
1025 t_9dmtogyjAocFsNW1igag 04:00:00 22:30:00 ... 22:30:00 18.5 18.0
1026 uWbmDMJcuxVgryNihVx3rw 06:00:00 23:00:00 ... 23:00:00 17.0 17.0
1027 F9v9bdFqp-GxMEjABRoDyA 04:30:00 21:30:00 ... 21:30:00 17.0 16.5
1028 RVUsquY9ozToz3Hhmohq4w 05:00:00 21:30:00 ... 21:30:00 16.5 16.0
1029 8S_6ltVRDCXp58xAAlVPAw 08:00:00 21:30:00 ... 21:30:00 13.5 12.5
1030 cwiEry81jficSY3ZrhgWvg 00:00:00 00:00:00 ... 00:00:00 0.0 0.0
1031 HPmQPrWMnPaMDlddiNpkBw 06:30:00 22:30:00 ... 22:30:00 16.0 15.5
1032 gqo-fkQyjs4ojJnDIsJyiQ 06:30:00 21:00:00 ... 21:00:00 14.5 15.5
1033 RHruCsEnWgOhXxgv5xNvEQ 04:30:00 21:30:00 ... 21:30:00 17.0 17.0
1034 9Nku1KO45GfQwEvXnFl8xA 05:00:00 21:00:00 ... 21:00:00 16.0 15.5
1035 5rFXq7AGqJJeNn48PHfEtA 00:00:00 00:00:00 ... 00:00:00 0.0 0.0
1036 gaGhS-nnnjW_i6WwPWH75A 00:00:00 00:00:00 ... 00:00:00 0.0 0.0
1037 u0Kzq5WYm0z29PTRAenzCQ 06:30:00 23:00:00 ... 23:00:00 16.5 15.5
1038 XBBGRLsPIVeaERwzW4xzGQ 04:00:00 23:00:00 ... 23:00:00 19.0 19.0
1039 7OuY7NAB1KuC9cYn8He3Pw 00:00:00 00:00:00 ... 00:00:00 0.0 0.0
1

In [40]:
%python # Table for multiple variable analysis
hour_diff_df = Hour_df.loc[:,['satdiff','sundiff']]
hour_diff_df

Out[ 8 ]: 
 satdiff sundiff
0 15.5 15.5
1 0.0 0.0
2 15.0 15.0
3 16.5 15.5
4 16.0 16.0
5 17.0 17.0
6 16.0 16.0
7 17.0 17.0
8 15.5 15.5
9 10.0 8.0
10 15.0 15.0
11 13.0 13.0
12 18.0 18.0
13 0.0 0.0
14 7.5 7.5
15 0.0 0.0
16 13.0 13.0
17 15.0 15.0
18 0.0 0.0
19 13.5 13.0
20 17.0 17.0
21 0.0 0.0
22 15.5 15.0
23 14.0 10.5
24 17.5 17.5
25 0.0 0.0
26 0.0 0.0
27 18.0 18.0
28 18.5 18.0
29 17.5 16.5
... ... ...
1012 14.5 14.5
1013 14.0 14.0
1014 16.0 15.0
1015 15.0 15.0
1016 0.0 0.0
1017 0.0 0.0
1018 14.5 14.5
1019 0.0 0.0
1020 16.5 16.5
1021 14.0 14.0
1022 14.0 14.0
1023 16.0 16.0
1024 16.0 16.0
1025 18.5 18.0
1026 17.0 17.0
1027 17.0 16.5
1028 16.5 16.0
1029 13.5 12.5
1030 0.0 0.0
1031 16.0 15.5
1032 14.5 15.5
1033 17.0 17.0
1034 16.0 15.5
1035 0.0 0.0
1036 0.0 0.0
1037 16.5 15.5
1038 19.0 19.0
1039 0.0 0.0
1040 16.0 16.0
1041 0.0 0.0

[1042 rows x 2 columns]

In [41]:
%python
hour_diff_df = hour_diff_df.stack()\
  .rename_axis(['num','weekend'],axis=0)\
  .to_frame()\
  .reset_index()\
  .rename(columns={0:'diff'}) \
  .drop(axis=1, labels=['num'])
  
hour_diff_df

Out[ 9 ]: 
 weekend diff
0 satdiff 15.5
1 sundiff 15.5
2 satdiff 0.0
3 sundiff 0.0
4 satdiff 15.0
5 sundiff 15.0
6 satdiff 16.5
7 sundiff 15.5
8 satdiff 16.0
9 sundiff 16.0
10 satdiff 17.0
11 sundiff 17.0
12 satdiff 16.0
13 sundiff 16.0
14 satdiff 17.0
15 sundiff 17.0
16 satdiff 15.5
17 sundiff 15.5
18 satdiff 10.0
19 sundiff 8.0
20 satdiff 15.0
21 sundiff 15.0
22 satdiff 13.0
23 sundiff 13.0
24 satdiff 18.0
25 sundiff 18.0
26 satdiff 0.0
27 sundiff 0.0
28 satdiff 7.5
29 sundiff 7.5
... ... ...
2054 satdiff 17.0
2055 sundiff 16.5
2056 satdiff 16.5
2057 sundiff 16.0
2058 satdiff 13.5
2059 sundiff 12.5
2060 satdiff 0.0
2061 sundiff 0.0
2062 satdiff 16.0
2063 sundiff 15.5
2064 satdiff 14.5
2065 sundiff 15.5
2066 satdiff 17.0
2067 sundiff 17.0
2068 satdiff 16.0
2069 sundiff 15.5
2070 satdiff 0.0
2071 sundiff 0.0
2072 satdiff 0.0
2073 sundiff 0.0
2074 satdiff 16.5
2075 sundiff 15.5
2076 satdiff 19.0
2077 sundiff 19.0
2078 satdiff 0.0
2079 sundiff 0.0
2080 satdiff 16.0
2081 sundiff 16.0
2082 satdiff 0.0
2083 sundiff 0.0

[2084 rows x 2 columns]

**(10) Create `park_info` view without null parktype from `starbucks` view**

In [43]:
%sql
drop view if exists park_info;
create temporary view park_info
as select state, attributes.BusinessParking as parktype from starbucks
where attributes.BusinessParking is not NULL